core-processor/src/common.rs

```diff
@@ -285,18 +294,30 @@ pub enum ExecutionErrorReason {
     /// Not enough gas to load memory
     #[display(fmt = "Not enough gas to load memory")]
     LoadMemoryGasExceeded,
+    /// Not enough gas in block to load memory
+    #[display(fmt = "Not enough gas in block to load memory")]
+    LoadMemoryBlockGasExceeded,
     /// Not enough gas to grow memory size
     #[display(fmt = "Not enough gas to grow memory size")]
     GrowMemoryGasExceeded,
-    /// Not enough gas for initial memory
-    #[display(fmt = "Not enough gas for initial memory")]
+    /// Not enough gas in block to grow memory size
+    #[display(fmt = "Not enough gas in block to grow memory size")]
+    GrowMemoryBlockGasExceeded,
+    /// Not enough gas for initial memory handling
+    #[display(fmt = "Not enough gas for initial memory handling")]
     InitialMemoryGasExceeded,
+    /// Not enough gas in block for initial memory handling
+    #[display(fmt = "Not enough gas in block for initial memory handling")]
+    InitialMemoryBlockGasExceeded,
```

Comment:
How errors `...GasExceeded` and `...BlockGasExceeded` differ? If I'm not confused there is no other gas except one in block.

Reply:
There is one gas counter for contract and another gas counter for block.